### PR TITLE
FEAT: Chat vl web UI

### PR DESCRIPTION
**Pull request by codingl2k1**
- Add web UI for multimodal model:
  - Launch UI.
  - Running model UI.
  - Chat vl gradio UI.
- Add gptq Int4 for qwen-vl-chat.
- Allow gptq model select multiple GPUs.

### CODE REVIEW

File: xinference/core/chat_interface.py [hunk 1]
```diff
@@ -12,31 +12,36 @@
...
 # limitations under the License.
 
+import base64
 import logging
 import os
+from io import BytesIO
 from typing import Generator, List, Optional
 
 import gradio as gr
+import PIL.Image
```
Comment (ChengjieLi28) [inline xinference/core/chat_interface.py:22]: Add this package to required dependencies

File: xinference/core/chat_interface.py [hunk 2]
```diff
@@ -173,6 +181,130 @@ def generate_wrapper(
             analytics_enabled=False,
         )
 
+    def build_chat_vl_interface(
+        self,
+    ) -> "gr.Blocks":
+        def predict(history, bot):
+            logger.debug("Predict.")
```
Comment (ChengjieLi28) [inline xinference/core/chat_interface.py:188]: Is this a duplicated logger when you test?

File: xinference/core/chat_interface.py [hunk 3]
```diff
@@ -173,6 +181,130 @@ def generate_wrapper(
...
+            from ..client import RESTfulClient
+
+            client = RESTfulClient(self.endpoint)
+            model = client.get_model(self.model_uid)
```
Comment (ChengjieLi28) [inline xinference/core/chat_interface.py:192]: Invoke set_token when authenticated.

File: xinference/web/ui/package-lock.json [hunk 4]
```diff
@@ -27,7 +27,6 @@
         "@testing-library/react": "^13.4.0",
```
Comment (ChengjieLi28) [inline xinference/web/ui/package-lock.json:27]: The changes in this file should be reverted.

File: xinference/core/chat_interface.py (hunk 1 above)
Comment (codingl2k1) [reply xinference/core/chat_interface.py:22]: I added pillow to the install requires. Thanks.

File: xinference/core/chat_interface.py (hunk 3 above)
Comment (codingl2k1) [reply xinference/core/chat_interface.py:193]: Fixed.

File: xinference/core/chat_interface.py (hunk 2 above)
Comment (codingl2k1) [reply xinference/core/chat_interface.py]: Fixed.

File: xinference/web/ui/package-lock.json (hunk 4 above)
Comment (codingl2k1) [reply xinference/web/ui/package-lock.json]: Fixed.

**Review by ChengjieLi28 — APPROVED** [review]

**Review by aresnow1 — APPROVED** [review]

**Pull request merged**